### ISSUE TITLE
Regression: Pillar fails to render when import'ing associated files, TemplateNotFound

### ISSUE THREAD

**Issue by davidjb** [saltstack/salt#19546]
My ``salt-ssh`` highstate calls now fail to render Pillar data that is using a Jinja ``import`` statement, because the given imported files can't be found.   Essentially, I have some files next to a given pillar, and in the past have been able to ``import`` them for use in the file.  This has been helpful in that I'm able to keep secret files (certs, keys) in a older, exclude that from version control, yet still have the data included in the Pillar when delivered.  The example configuration is given in the error traceback below.  

This appears to be a regression somewhere between ``v2014.7.0rc7`` and the current ``2014.7`` or ``2015.2`` branches.  Using the exact same configuration, the former version works correctly, and the latter branches both fail with ``TemplateNotFound`` errors, as follows:

```
$ salt-ssh '*' state.highstate
nginx:
  {% import 'private/hostname.crt' as ssl_cert %}    <======================
  certificate: |-
    {{ ssl_cert|string|indent(4) }}
[...]
---
[CRITICAL] Pillar render error: Rendering SLS 'base' failed. Please see master log for details.
[CRITICAL] Rendering SLS 'base' failed, render error:
Jinja error: private/hostname.crt
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/salt/utils/templates.py", line 291, in render_jinja_tmpl
    output = template.render(**unicode_context)
  File "/usr/local/lib/python2.7/dist-packages/jinja2/environment.py", line 894, in render
    return self.environment.handle_exception(exc_info, True)
  File "<template>", line 21, in top-level template code
  File "/usr/local/lib/python2.7/dist-packages/salt/utils/jinja.py", line 130, in get_source
    raise TemplateNotFound(template)
TemplateNotFound: private/hostname.crt
```

**Comment by davidjb** [comment]
FYI, versions for working:
```
salt-ssh --versions-report
           Salt: 2014.7.0rc7
         Python: 2.7.8 (default, Oct 20 2014, 15:05:19)
         Jinja2: 2.6
       M2Crypto: 0.21.1
 msgpack-python: 0.4.2
   msgpack-pure: Not Installed
       pycrypto: 2.6.1
        libnacl: Not Installed
         PyYAML: 3.11
          ioflo: Not Installed
          PyZMQ: 14.3.1
           RAET: Not Installed
            ZMQ: 4.0.4
           Mako: 1.0.0
```

and not working:
```
salt-ssh --versions-report
           Salt: 2014.7.0-1242-g2623aa6
         Python: 2.7.8 (default, Oct 20 2014, 15:05:19)
         Jinja2: 2.6
       M2Crypto: 0.21.1
 msgpack-python: 0.4.2
   msgpack-pure: Not Installed
       pycrypto: 2.6.1
        libnacl: Not Installed
         PyYAML: 3.11
          ioflo: Not Installed
          PyZMQ: 14.3.1
           RAET: Not Installed
            ZMQ: 4.0.4
           Mako: 1.0.0
```

**Comment by davidjb** [comment]
So the issue is this commit:  https://github.com/saltstack/salt/commit/fd604d1abaee20db133a684d861f6bd463ef1663

Which looks to have been applied, reverted, and then applied again?  I'm confused: https://github.com/saltstack/salt/commits/2014.7/salt/utils/jinja.py

**Comment by davidjb** [comment]
...and reverting 1e63b698474a141039d2ae2f7aaa778fd5e7884b (the last re-application of the commit) solves my problem. Not sure how this affects #17963.

**Comment by davidjb** [comment]
#19552 is going to fix the problem.

**Comment by davidjb** [comment]
Now fixed with #19552 being merged.  Closing.

**Comment by rallytime** [comment]
Thanks for the report and glad we got that fixed up so quickly! Thanks for following up!
